src/domain/mongoose_domain_sql.erl

```diff
@@ -138,34 +143,51 @@ select_from(FromId, Limit) ->
 select_updates_from(FromId, Limit) ->
     Pool = get_db_pool(),
     Args = rdbms_queries:add_limit_arg(Limit, [FromId]),
-    {selected, Rows} = execute_successfully(Pool, domain_select_events_from, Args),
-    Rows.
-
-get_min_event_id() ->
-    Pool = get_db_pool(),
-    selected_to_int(execute_successfully(Pool, domain_events_min, [])).
+    case execute_successfully(Pool, domain_select_events_from, Args) of
+        {selected, []} ->
+            %% get MaxID, this inserts the dummy record
+            %% in the events table if required.
+            MaxId = get_max_event_id_or_set_dummy(),
+            if
+                MaxId > FromId ->
+                    select_updates_from(FromId, Limit);
```

Comment:
I don't want to edit comments, but I've just realized that that it might be actually possible to get **more** than just the dummy row there if case another node made an update to the table, so I edited the suggestion above.